lib/pgBackRest/Backup/Backup.pm

```diff
@@ -935,43 +932,32 @@ sub process
         my @stryArchive = lsnFileRange($strLsnStart, $strLsnStop, $strDbVersion);
 
         foreach my $strArchive (@stryArchive)
         {
-            my $strArchiveFile = walSegmentFind($oFileLocal, $strArchiveId, $strArchive, optionGet(OPTION_ARCHIVE_TIMEOUT));
+            my $strArchiveFile = walSegmentFind($oStorageRepo, $strArchiveId, $strArchive, optionGet(OPTION_ARCHIVE_TIMEOUT));
             $strArchive = substr($strArchiveFile, 0, 24);
 
             if (optionGet(OPTION_BACKUP_ARCHIVE_COPY))
             {
                 logDebugMisc($strOperation, "archive: ${strArchive} (${strArchiveFile})");
 
                 # Copy the log file from the archive repo to the backup
-                my $strDestinationFile = MANIFEST_TARGET_PGDATA . "/pg_xlog/${strArchive}" .
-                                         ($bCompress ? ".$oFileLocal->{strCompressExtension}" : '');
-                my $bArchiveCompressed = $strArchiveFile =~ "^.*\.$oFileLocal->{strCompressExtension}\$";
+                my $bArchiveCompressed = $strArchiveFile =~ ('^.*\.' . COMPRESS_EXT . '\$');
 
-                my ($bCopyResult, $strCopyChecksum, $lCopySize) =
-                    $oFileLocal->copy(PATH_BACKUP_ARCHIVE, "${strArchiveId}/${strArchiveFile}",
-                                 PATH_BACKUP_CLUSTER, "${strBackupLabel}/${strDestinationFile}",
-                                 $bArchiveCompressed, $bCompress,
-                                 undef, $lModificationTime, undef, true,
-                                 undef, undef, undef, undef, undef, undef, false);
+                $oStorageRepo->copy(
+                    STORAGE_REPO_ARCHIVE . "/${strArchiveId}/${strArchiveFile}",
+                    STORAGE_REPO_BACKUP . "/${strBackupLabel}/" . MANIFEST_TARGET_PGDATA . "/pg_xlog/${strArchive}" .
+                        ($bCompress ? qw{.} . COMPRESS_EXT : ''));
 
                 # Add the archive file to the manifest so it can be part of the restore and checked in validation
                 my $strPathLog = MANIFEST_TARGET_PGDATA . '/pg_xlog';
                 my $strFileLog = "${strPathLog}/${strArchive}";
 
-                # Compare the checksum against the one already in the archive log name
-                if ($strArchiveFile !~ "^${strArchive}-${strCopyChecksum}(\\.$oFileLocal->{strCompressExtension}){0,1}\$")
-                {
-                    confess &log(ERROR, "error copying WAL segment '${strArchiveFile}' to backup - checksum recorded with " .
-                                        "file does not match actual checksum of '${strCopyChecksum}'", ERROR_CHECKSUM);
-                }
-
                 # Add file to manifest
-                $oBackupManifest->fileAdd($strFileLog, $lModificationTime, $lCopySize, $strCopyChecksum, true);
+                $oBackupManifest->fileAdd($strFileLog, $lModificationTime, 16777216, substr($strArchiveFile, 25, 40), true);
```

Comment:
Create a constant (PG_WAL_SEGMENT_SIZE) - we should do something more clever here in the future but this will do for now.